src/messages.ts

```diff
@@ -16,8 +16,12 @@
 
 export const PROBLEMATIC_CREDENTIALS_WARNING =
     `Your application has authenticated using end user credentials from Google
-  Cloud SDK. We recommend that most server applications use service accounts
-  instead. If your application continues to use end user credentials from Cloud
-  SDK, you might receive a "quota exceeded" or "API not enabled" error. For
-  more information about service accounts, see
-  https://cloud.google.com/docs/authentication/.`;
+Cloud SDK. We recommend that most server applications use service accounts
+instead. If your application continues to use end user credentials from Cloud
+SDK, you might receive a "quota exceeded" or "API not enabled" error. For
+more information about service accounts, see
+https://cloud.google.com/docs/authentication/.`;
+
+export const DEFAULT_PROJECT_ID_DEPRECATED =
+    `The 'getDefaultProjectId' method has been deprecated, and will be removed
+ in the 3.0 release of this library. Please use the 'getProjectId' method instead.`;
```

Comment:
The warning will reach the end user and it may not be clear what 'this library' means. It would be just a disembodied warning message without any context. Maybe spell out the library name instead?

Secondly, apart from a small number users who are using this library directly, most users of this library get this as a dependency of another Google Cloud client library. For those use-cases, I think it is a fatal error if we client library uses a deprecated method, and we should try out best to ensure this warning doesn't leak to the end user.  Are we doing anything to ensure that the client libraries do not end up using any deprecated APIs?

Reply:
I updated this to use `google-auth-library` instead :) On the second point - should we run the system tests with `--throw-deprecations` flag?